plugin/clone/src/clone_server.cc

```diff
@@ -380,8 +380,10 @@ int Server::deserialize_init_buffer(const uchar *init_buf, size_t init_len) {
     if (init_len < serialized_length) {
       goto err_end;
     }
-
-    m_storage_vec.push_back(loc);
+    
+    if (loc.m_hton != nullptr) {
```

Comment:
When client is rocksdb and donor is innodb. The client sends rocksdb locator to donor but the donor can not serialize it with valid m_hton. It will cause nullptr error later, so I only add loc to storage_vector when m_hton is not null. Looking for better solution.

Reply:
The clone will actually fail with ```ERROR 3863 (HY000): Clone received unexpected response from Donor : Wrong Clone RPC response length for COM_RES_LOCS.``` if I remove the loc here.

Reply:
Can you just do `my_error(ER_CLONE_PROTOCOL, MYF(0), "SE mismatch")` when `loc.m_hton = nullptr`?